app/src/test/java/com/codesoom/assignment/application/UserCommandServiceTest.java

```diff
@@ -0,0 +1,250 @@
+            }
+        }
+
+        @Nested
+        @DisplayName("저장되어있지 않은 user 의 id가 주어지면 ")
+        class Context_with_non_existence_user_id {
+            @Test
+            @DisplayName("사용자가 존재하지 않다는 예외를 던진다")
+            void it_throws_exception() {
+                assertThatThrownBy(
+                        () -> userCommandService.deleteUser(INVALID_USER_ID)
+                ).isExactlyInstanceOf(UserNotFoundException.class);
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("deleteUsers 메서드는")
+    @SpringBootTest
+    class Describe_deleteUsers {
+
+        @Autowired
+        private UserRepository userRepository;
+
+        @Nested
+        @DisplayName("저장되어있는 user 의 id가 주어지면 ")
+        class Context_with_existing_user_id {
+            private Set<Long> deletedIds;
+
+            @BeforeEach
+            void setUp() {
+                UserResponse savedUser1 = userCommandService.createUser(
+                        UserCreateRequest.builder()
+                                .email("a@a.com")
+                                .name("김 코")
+                                .password("a")
+                                .build()
+                );
+                UserResponse savedUser2 = userCommandService.createUser(
+                        UserCreateRequest.builder()
+                                .email("b@b.com")
+                                .name("김 코")
+                                .password("b")
+                                .build()
+                );
+
+                deletedIds = Set.of(savedUser1.getId(), savedUser2.getId());
+            }
+
+            @Test
+            @DisplayName("user 들을 삭제하고 삭제한 user ids 와 삭제하지 못한 user ids(빈 set)를 리턴한다")
+            void it_returns_delete_user_report() {
+
+                UserDeleteReport userDeleteReport = userCommandService.deleteUsers(deletedIds);
+
+                assertThat(userRepository.findAllById(deletedIds)).isEmpty();
+                assertThat(userRepository.findAllById(userDeleteReport.getDeletedSuccessIds())).isEmpty();
+
+                assertThat(userDeleteReport.getDeletedSuccessIds()).isEqualTo(deletedIds);
+                assertThat(userDeleteReport.getDeletedFailIds()).isEmpty();
+            }
+        }
+
+        @Nested
+        @DisplayName("저장되어있지 않은 user 의 id가 주어지면 ")
+        class Context_with_non_existence_user_id {
+            private Long deleteId;
+
+            @BeforeEach
+            void setUp() {
+                UserResponse savedUser = userCommandService.createUser(
+                        UserCreateRequest.builder()
+                                .email("a@a.com")
+                                .name("김 코")
+                                .password("a")
+                                .build()
+                );
+                deleteId = savedUser.getId();
+            }
+
+            @Test
+            @DisplayName("user 들을 삭제하고 삭제한 user ids 와 삭제하지 못한 user ids 를 리턴한다")
```

Comment:
테스트 메시지가 보고서의 내용에 의존하고 있습니다.
```suggestion
            @DisplayName("user 들을 삭제하고 삭제 보고서를 리턴한다")
```

Reply:
0f96da4d9b672a98bf4914af31d2fb3117e7e4f0 수정했습니다!